pkg/identity/cache/allocator.go

```diff
@@ -437,5 +437,16 @@ func (m *CachingIdentityAllocator) ReleaseSlice(ctx context.Context, owner Ident
 // syncs all identities to the local identity cache.
 func (m *CachingIdentityAllocator) WatchRemoteIdentities(backend kvstore.BackendOperations) *allocator.RemoteCache {
 	<-m.globalIdentityAllocatorInitialized
-	return m.IdentityAllocator.WatchRemoteKVStore(backend, m.identitiesPath)
+
+	remoteAllocatorBackend, err := kvstoreallocator.NewKVStoreBackend(m.identitiesPath, m.owner.GetNodeSuffix(), GlobalIdentity{}, backend)
+	if err != nil {
+		log.WithError(err).WithField(logfields.BackendName, backend).Error("Error setting up remote allocator backend")
```

Comment:
Shouldn't this return nil here?

Reply:
Yes, but it turned out to be worse. The caller of this didn't handle the nil case either. I made this return an error that is handled in the controller.